modules/shared/src/main/scala/org/tessellation/ext/kryo.scala

```diff
@@ -33,4 +36,19 @@ object kryo {
       KryoSerializer[F].deserialize[A](bytes).liftTo[F]
   }
 
+  type KryoRegistrationId[C] = Int Refined C
+
+  def union[C1, C2](
+    a: Map[Class[_], KryoRegistrationId[C1]],
+    b: Map[Class[_], KryoRegistrationId[C2]]
+  ): Map[Class[_], KryoRegistrationId[C1 Or C2]] =
+    a.view.mapValues { x =>
+      val y: KryoRegistrationId[C1 Or C2] = x
+      y
+    }.toMap ++
+      b.view.mapValues { x =>
+        val y: KryoRegistrationId[C1 Or C2] = x
+        y
+      }.toMap
```

Comment:
try this:
```scala
a.view.mapValues[KryoRegistrationId[C1 Or C2]](identity) ++
  b.view.mapValues[KryoRegistrationId[C1 Or C2]](identity)
```

Reply:
It won't compile.

Reply:
@tbekas can you verify that?
it compiles correctly on my end
![image](https://user-images.githubusercontent.com/4959931/157498157-7a8707e4-2d1c-446b-9f91-9cf4d0b934a9.png)